zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/MigrateProcessInstanceTest.java

```diff
@@ -1091,6 +1094,164 @@ public void shouldUnsubscribeUnmappedMessageBoundaryEvent() {
+        .isPresent();
+  }
+
+  @Test
+  public void shouldWriteMigratedEventWhenActiveElementHasAProcessIncident() {
+    // given
+    final String sourceProcessId = "process";
+    final String targetProcessId = "process2";
+    final var deployment =
+        ENGINE
+            .deployment()
+            .withXmlResource(
+                Bpmn.createExecutableProcess(sourceProcessId)
+                    .startEvent()
+                    .serviceTask(
+                        "A",
+                        b ->
+                            b.zeebeJobType("jobTypeA")
+                                .zeebeInputExpression("assert(x, x != null)", "y"))
+                    .endEvent()
+                    .done())
+            .withXmlResource(
+                Bpmn.createExecutableProcess(targetProcessId)
+                    .startEvent()
+                    .serviceTask("B", t -> t.zeebeJobType("jobTypeB"))
+                    .endEvent("target_process_end")
+                    .done())
+            .deploy();
+    final long processInstanceKey =
+        ENGINE.processInstance().ofBpmnProcessId(sourceProcessId).create();
+
+    final Record<IncidentRecordValue> incident =
+        RecordingExporter.incidentRecords(IncidentIntent.CREATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .getFirst();
+
+    final long targetProcessDefinitionKey =
+        extractProcessDefinitionKeyByProcessId(deployment, targetProcessId);
+
+    // when
+    ENGINE
+        .processInstance()
+        .withInstanceKey(processInstanceKey)
+        .migration()
+        .withTargetProcessDefinitionKey(targetProcessDefinitionKey)
+        .addMappingInstruction("A", "B")
+        .migrate();
+
+    // then
+    assertThat(
+            RecordingExporter.incidentRecords(IncidentIntent.MIGRATED)
+                .withProcessInstanceKey(processInstanceKey)
+                .getFirst()
+                .getValue())
+        .describedAs("Expect that process definition is updated")
+        .hasProcessDefinitionKey(targetProcessDefinitionKey)
+        .hasBpmnProcessId(targetProcessId)
+        .describedAs("Expect that element id changed due to mapping")
+        .hasElementId("B");
+
+    // after resolving the incident, the process should continue as expected
+    ENGINE.variables().ofScope(processInstanceKey).withDocument(Map.of("x", 1)).update();
+    ENGINE.incident().ofInstance(processInstanceKey).withKey(incident.getKey()).resolve();
+
+    assertThat(
+            RecordingExporter.jobRecords(JobIntent.CREATED)
+                .withProcessInstanceKey(processInstanceKey)
+                .withElementId("B") // target element id
+                .exists())
+        .describedAs("Expect that the job is created in the migrated process")
+        .isTrue();
+
+    // since the job wasn't created in the source process before the migration, a new job is created
+    // in the target process after resolving the incident
```

Comment:
Yeah, that was belong to above assertion and somehow ended up being on top of the job complete. 🤷‍♂️ I applied your suggestion but that increased the test duration from 400ms to 5 seconds. Is it because waiting element completion takes time? @korthout

I added 2 separate assertions for now. 👍

Reply:
Since we now assert incident resolved fields to verify the incident migration updated in the state, I think, should be fine to keep both assertions separate.